autogen/main.tf

```
@@ -38,6 +38,7 @@ resource "random_shuffle" "available_zones" {
 locals {
   // location
   location = var.regional ? var.region : var.zones[0]
+  region   = var.region ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

Condition values can not be `null` so checking `var.region` directly is invalid.

```suggestion
  region   = var.region != null ? var.region : join("-", slice(split("-", var.zones[0]), 0, 2))
```

---

fixed